### PR TITLE
Fix default comparator for strings

### PR DESCRIPTION
**Pull request by MaxGraey**
Default comparator was not tested well. So now use actually `comparator<string>()` instead external `(a, b) => (a > b) - (a < b)` and fix special cases for this comparison.

### CODE REVIEW

File: std/assembly/internal/sort.ts [hunk 1]
```diff
@@ -36,8 +36,12 @@ export function COMPARATOR<T>(): (a: T, b: T) => i32 {
     }
   } else if (isString<T>()) {
     return (a: T, b: T): i32 => {
-      var sa = <string>a, sb = <string>b;
-      return compareUnsafe(sa, 0, sb, 0, min(sa.length, sb.length));
+      if (a === b || a === null || b === null) return 0;
+      var alen = (<string>a).length;
+      var blen = (<string>b).length;
+      if (!alen) return -1;
```
Comment (MaxGraey) [inline std/assembly/internal/sort.ts]: Not sure about `if (alen == alen && alen == 0) return 0`. This works well without this check, but potentially make sorting unstable.

Comment (dcodeIO) [reply std/assembly/internal/sort.ts]: Not sure about the additional case, seems that we don't need it. Or does it prevent issues?

Comment (MaxGraey) [reply std/assembly/internal/sort.ts]: it potentially prevent unstable sorting when we compare one empty string `""` with another empty string and they are could potentially swap and change order. Currently all browser engines switches to stable sort based on TimSort and I also have plan rewrite existing sort algorithm to TimSort or GrailSort (both stable)

**Comment by dcodeIO** [comment]
Thanks! :)

**Pull request merged**